glue/core/data.py

```diff
@@ -238,6 +242,55 @@ def __str__(self):
     def jitter(self, method=None):
         raise NotImplementedError
 
+    def subset_from_roi(self, att, roi, other_comp=None, other_att=None, coord='x'):
+        """ Create a SubsetState object from an ROI.
+
+        This encapsulates the logic for creating subset states with
+        Components. See the documentation for CategoricalComponents for caveats
+        involved with mixed-type plots.
+
+        :param att: attribute name of this Component
+        :param roi: an ROI object
+        :param other_comp: The other Component for 2D ROIs
+        :param other_att: The attribute name of the other Component
+        :param coord: The orientation of this Component
+        :param is_nested: True if this was passed from another Component.
+        :return: A SubsetState (or subclass) object
+        """
+
+        assert coord in set('xy')
```

Comment:
minor nit: converting to set probably adds more overhead than it saves :). Also, this should be a ValueError and not an AssertionError, since it could be user-triggered